src/payment/strategies/affirm/affirm-payment-strategy.spec.ts

```
@@ -0,0 +1,196 @@
+        });
+    });
+
+    describe('#execute()', () => {
+        beforeEach(async () => {
+            await strategy.initialize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway });
+
+            jest.spyOn(affirm.checkout, 'open').mockImplementation(({ onSuccess }) => {
+                onSuccess({
+                    checkout_token: '1234',
+                    created: '1234',
+                });
+            });
+        });
+
+        it('creates order, checkout and payment', async () => {
+            jest.spyOn(store, 'dispatch');
+            const options = { methodId: 'affirm', gatewayId: undefined };
+
+            await strategy.execute(payload, options);
+
+            expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalledWith({ useStoreCredit: false }, options);
+            expect(affirm.checkout).toHaveBeenCalled();
+            expect(affirm.checkout.open).toHaveBeenCalled();
+            expect(affirm.ui.error.on).toHaveBeenCalled();
+            expect(store.dispatch).toHaveBeenCalledWith(submitPaymentAction);
+            expect(paymentActionCreator.submitPayment).toBeCalledWith({
+                methodId: paymentMethod.id,
+                paymentData: { nonce: '1234' },
+            });
+        });
+
+        it('returns error on affirm if users cancel flow', async () => {
+            jest.spyOn(affirm.checkout, 'open').mockImplementation(({ onFail }) => {
+                onFail();
+            });
+            await expect(strategy.execute(payload)).rejects.toThrow(PaymentMethodCancelledError);
+        });
+
+        it('does not create order/payment if paymentId is not set', () => {
+            payload.payment = undefined;
+            const error = 'Unable to submit payment for the order because the payload is invalid. Make sure the following fields are provided correctly: payment.methodId.';
```

Should actually throw an error instead of a string.